### ISSUE TITLE
Setext headers which start like ordered lists are parsed as a list item

### ISSUE THREAD

**Issue by KyleAMathews** [remarkjs/remark#218]
E.g.
```
1. first header
==========

2. second header
---------------
```

becomes

```
1.  # first header

2.  second header

* * *
```

**Comment by wooorm** [comment]
Could you expand on what you expect?

Here’s how GH, markdown-it, and CommonMark produce:

1. first header ==========

2. second header
---------------

Whereas remark produces:

1. first header
   ==========

2. second header

---------------

remark is definitely outnumbered here, but from what I see the others do this shouldn’t be headers!

**Comment by KyleAMathews** [comment]
Huh, weird. I assumed that the header syntax would take precedent over the list syntax so they'd both become headers e.g. `<h1>1. first header</h1>`. Which is what happens with ATX syntax. I'd assumed the two should work the same.

I was working on a project where I was converting html to markdown using pandoc then doing some extra refinements using Remark's AST but ran into this issue. Telling pandoc to always use ATX solved things I guess.

**Comment by wooorm** [comment]
Sorry for the wait! I just wanted to share that there’s now a PR that solves this issue: https://github.com/remarkjs/remark/pull/536.

**Comment by wooorm** [comment]
This is now released in [`remark@13.0.0`](https://github.com/remarkjs/remark/releases/tag/13.0.0)
